tests/testsuite/freshness_checksum.rs

```diff
@@ -0,0 +1,2880 @@
+//! Tests for fingerprinting (rebuild detection).
+
+#![allow(deprecated)]
+
+use std::fs::{self, OpenOptions};
+use std::io::prelude::*;
+use std::net::TcpListener;
+use std::process::Stdio;
+use std::thread;
+
+use super::death;
+use cargo_test_support::paths::CargoPathExt;
+use cargo_test_support::prelude::*;
+use cargo_test_support::registry::Package;
+use cargo_test_support::{basic_lib_manifest, basic_manifest, project, rustc_host, rustc_host_env};
+
+#[cargo_test]
+fn checksum_actually_uses_checksum() {
```

Comment:
At minimum, could you structure your PR so its
- A commit with these tests without `-Zchecksum-freshness`
- A commit with the checksum work that also updates the tests to pass `-Zchecksum-freshness`

A big benefit to this is it shows to reviewers / the community how this feature is comparing to what was being done before

(sometimes, I also break out "adding an unstable feature" into its own commit which is the flag + docs)

Reply:
Note: I've not dug deep into the tests, waiting on this change

Reply:
So it's worth noting that `freshness_checksum.rs` derives very heavily from `freshness.rs`. So a version of `freshness_checksum.rs` without the freshness flag would just be a subset of `freshness.rs`. I'm not sure how this provides new information. There are two tests which are truly unique to `freshness_checksum.rs`, which are `same_size_different_content()` and `checksum_actually_uses_checksum()`.

One might debate the merit of duplicating the tests like that. If you really wanted to deduplicate the tests then this would likely require a special case be added to the test runner code.